vendor.conf

```diff
@@ -42,3 +42,35 @@ github.com/stevvooe/ttrpc d4528379866b0ce7e9d71f3eb96f0582fc374577
 github.com/syndtr/gocapability db04d3cc01c8b54962a58ec7e491717d06cfcc16
 github.com/gotestyourself/gotestyourself 44dbf532bbf5767611f6f2a61bded572e337010a
 github.com/google/go-cmp v0.1.0
+# cri dependencies
+github.com/containerd/cri-containerd 83573155645882403ef2da86df8ea95e2d031f28
+github.com/blang/semver v3.1.0
+github.com/containernetworking/cni v0.6.0
+github.com/containernetworking/plugins v0.6.0
+github.com/cri-o/ocicni 9b451e26eb7c694d564991fbf44f77d0afb9b03c
+github.com/davecgh/go-spew v1.1.0
+github.com/docker/distribution b38e5838b7b2f2ad48e06ec4b500011976080621
+github.com/docker/docker 86f080cff0914e9694068ed78d503701667c4c00
+github.com/docker/spdystream 449fdfce4d962303d702fec724ef0ad181c92528
+github.com/emicklei/go-restful ff4f55a206334ef123e4f79bbf348980da81ca46
+github.com/fsnotify/fsnotify 7d7316ed6e1ed2de075aab8dfc76de5d158d66e1
+github.com/ghodss/yaml 73d445a93680fa1a78ae23a5839bad48f32ba1ee
+github.com/golang/glog 44145f04b68cf362d9c4df2182967c2275eaefed
+github.com/google/gofuzz 44d81051d367757e1c7c6a5a86423ece9afcf63c
+github.com/hashicorp/errwrap 7554cd9344cec97297fa6649b055a8c98c2a1e55
+github.com/hashicorp/go-multierror ed905158d87462226a13fe39ddf685ea65f1c11f
+github.com/json-iterator/go 1.0.4
+github.com/opencontainers/runtime-tools 6073aff4ac61897f75895123f7e24135204a404d
+github.com/opencontainers/selinux 4a2974bf1ee960774ffd517717f1f45325af0206
+github.com/seccomp/libseccomp-golang 32f571b70023028bd57d9288c20efbcb237f3ce0
+github.com/spf13/pflag v1.0.0
+github.com/tchap/go-patricia 5ad6cdb7538b0097d5598c7e57f0a24072adf7dc
+golang.org/x/time f51c12702a4d776e4c1fa9b0fabab841babae631
+gopkg.in/inf.v0 3887ee99ecf07df5b447e9b00d9c0b2adaa9f3e4
+gopkg.in/yaml.v2 53feefa2559fb8dfa8d81baad31be332c97d6c77
+k8s.io/api a1d6dce6736a6c75929bb75111e89077e35a5856
+k8s.io/apimachinery 8259d997cf059cd83dc47e5f8074b7a7d7967c09
+k8s.io/apiserver 8e45eac9dff86447a5c2effe6a3d2cba70121ebf
+k8s.io/client-go 33bd23f75b6de861994706a322b0afab824b2171
+k8s.io/kubernetes 05944b1d2ca7f60b09762a330425108f48f6b603
+k8s.io/utils 258e2a2fa64568210fbd6267cf1d8fd87c3cb86e
```

Comment:
So many dependencies! It's almost doubled.

I'm going to try and trace some of these down to see if we can remove any of them...

Reply:
Most of them are introduced by Kubernetes api, which might be hard to get rid of.

Reply:
kube depends on cri, cri depends on kube.  somethings never change :)

Reply:
CRI itself is relatively clean. :)
But the streaming server is not well cleaned up. :(

Reply:
The API machinery dependencies are crazy. We ran into this problem with `docker/cli` as well. I think the packages are not properly scoped, so you end up pulling in a bunch of unnecessary dependencies. 

For example:
https://github.com/kubernetes/apimachinery/blob/e33566cd44e02218ceb7d96b21ea8d03880e3ff5/pkg/apis/meta/internalversion/register.go#L41-L42

This exported `Codecs`, which is not used interally in this package at all, and seems to have [nothing to do with the rest of the package](https://godoc.org/github.com/kubernetes/apimachinery/pkg/apis/meta/internalversion) brings in a ton of unnecessary dependencies around yaml.

```
k8s.io/apimachinery/pkg/runtime/serializer ->
k8s.io/apimachinery/pkg/runtime/serializer/json ->
github.com/ghodss/yaml
```

https://godoc.org/k8s.io/apimachinery/pkg/util/httpstream/spdy is another case. It implements both client and server, which means anything that wants to create just a client or just a server is stuck with the dependencies for both. This also brings in yaml for no reason. It looks like this is used by the streaming server you were talking about.


I agree this is not easy to clean up, but I think it's necessary. I'll try submitting a PR and cc you.

Reply:
@dnephin Thanks a lot!

Reply:
Filed an issue. https://github.com/kubernetes/kubernetes/issues/60323
But I'm not sure how much we could improve after I read the code again. Let's try. :)
@dnephin